index.js

```diff
@@ -2,13 +2,21 @@
 
 const fp = require('fastify-plugin')
 const { createHash } = require('crypto')
+const { xxHash32 } = require('js-xxhash')
 const fnv1a = require('./fnv1a')
 
 function buildHashFn (algorithm = 'fnv1a') {
   if (algorithm === 'fnv1a') {
     return (payload) => '"' + fnv1a(payload).toString(36) + '"'
   }
 
+  if (algorithm === 'xxhash') {
+    return (payload) => {
+      const bufferPayload = payload instanceof Buffer ? payload : Buffer.from(payload, 'utf8')
```

Comment:
I prefer a buffer-specific algorithm instead. This is essentially the reason why I implemented fnv1a in this module.

Reply:
Do you mean that the library itself would internally handle both Strings and Buffers?

Reply:
Yes. Essentially we are losing a significant amount of throughput by having a string being converted to a buffer for the hashing. This is going to get problematic with longer payloads as well.

Reply:
Makes sense. I'll check your implementation better and see how to incorporate the original lib and do the necessary changes. Thanks for clarifying